delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,267 @@
+    */
+  final val moduleType: String = "resolver"
+
+  private[delta] val inProjectType    = "in-project"
+  private[delta] val crossProjectType = "cross-project"
+
+  import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils.instant
+  import io.scalaland.chimney.dsl._
+
+  private[delta] def next(state: ResolverState, event: ResolverEvent): ResolverState = {
+    (state, event) match {
+      // Creating an in-project resolver
+      case (Initial, e: InProjectResolverCreated)                           =>
+        e.into[InProjectCurrent]
+          .withFieldConst(_.deprecated, false)
+          .withFieldConst(_.createdAt, e.instant)
+          .withFieldConst(_.createdBy, e.subject)
+          .withFieldConst(_.updatedAt, e.instant)
+          .withFieldConst(_.updatedBy, e.subject)
+          .transform
+      // Updating an in-project resolver
+      case (c: InProjectCurrent, e: InProjectResolverUpdated)               =>
+        c.copy(priority = e.priority, rev = e.rev, updatedAt = e.instant, updatedBy = e.subject)
+
+      // Creating a cross-project resolver
+      case (Initial, e: CrossProjectResolverCreated)                        =>
+        e.into[CrossProjectCurrent]
+          .withFieldConst(_.deprecated, false)
+          .withFieldConst(_.createdAt, e.instant)
+          .withFieldConst(_.createdBy, e.subject)
+          .withFieldConst(_.updatedAt, e.instant)
+          .withFieldConst(_.updatedBy, e.subject)
+          .transform
+      // Updating a cross-project resolver
+      case (c: CrossProjectCurrent, e: CrossProjectResolverUpdated)         =>
+        c.copy(
+          priority = e.priority,
+          resourceTypes = e.resourceTypes,
+          projects = e.projects,
+          identities = e.identities,
+          rev = e.rev,
+          updatedAt = e.instant,
+          updatedBy = e.subject
+        )
+
+      // Deprecating a in-project resolver
+      case (c: InProjectCurrent, e: ResolverDeprecated) if !c.deprecated    =>
+        c.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+      // Deprecating a cross-project resolver
+      case (c: CrossProjectCurrent, e: ResolverDeprecated) if !c.deprecated =>
+        c.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+
+      // Cases that should happen and that should have been filtered out by the evaluation
+      case (s, _)                                                           => s
+    }
+  }
+
+  private[delta] def evaluate(
+      isOpen: ProjectRef => IO[ReadOnlyProject, Unit]
+  )(state: ResolverState, command: ResolverCommand)(implicit
+      caller: Caller,
+      clock: Clock[UIO]
+  ): IO[ResolverRejection, ResolverEvent] = {
+
+    // All provided identities must be part of the caller identities
+    def validateIdentities(identities: Set[Identity]): IO[ResolverRejection, Unit] = {
+      if (identities.isEmpty)
+        IO.raiseError(NoIdentities)
```

Comment:
I think we are going to allow empty identities in order to support [this](https://github.com/BlueBrain/nexus/issues/1661)

Reply:
I prefer to do everything in a separate PR

Reply:
Ok. Then add a TODO please, as I said in another comment

Reply:
I finally added it :)